tests/foreman/cli/test_setting.py

```diff
@@ -33,315 +34,352 @@
 from robottelo.decorators import run_in_one_thread
 from robottelo.decorators import tier2
 from robottelo.decorators import tier3
-from robottelo.test import CLITestCase
 
 
-class SettingTestCase(CLITestCase):
-    """Implements tests for Settings for CLI"""
+@pytest.mark.stubbed
+@tier2
+def test_negative_update_hostname_with_empty_fact():
+    """Update the Hostname_facts settings without any string(empty values)
+
+    :id: daca0746-75ad-42fb-97ed-6db0224eeeac
+
+    :expectedresults: Error should be raised on setting empty value for hostname_facts setting
+
+    :CaseAutomation: notautomated
+    """
+
+
+@tier2
+@pytest.mark.parametrize('setting_update', ['discovery_prefix'], indirect=True)
+def test_positive_update_hostname_prefix_without_value(setting_update):
+    """Update the Hostname_prefix settings without any string(empty values)
+
+    :id: a84c28ea-6821-4c31-b4ab-8662c22c9135
+
+    :expectedresults: Hostname_prefix should be set without any text
+
+    :parametrized: yes
+    """
+    Settings.set({'name': "discovery_prefix", 'value': ""})
+    discovery_prefix = Settings.list({'search': 'name=discovery_prefix'})[0]
+    assert discovery_prefix['value'] == ''
+
+
+@tier2
+@pytest.mark.parametrize('setting_update', ['discovery_prefix'], indirect=True)
+def test_positive_update_hostname_default_prefix(setting_update):
+    """Update the default set prefix of hostname_prefix setting
+
+    :id: a6e46e53-6273-406a-8009-f184d9551d66
+
+    :expectedresults: Default set prefix should be updated with new value
+
+    :parametrized: yes
+    """
+    hostname_prefix_value = gen_string('alpha')
+    Settings.set({'name': "discovery_prefix", 'value': hostname_prefix_value})
+    discovery_prefix = Settings.list({'search': 'name=discovery_prefix'})[0]
+    assert hostname_prefix_value == discovery_prefix['value']
+
+
+@pytest.mark.stubbed
+@tier2
+def test_positive_update_hostname_default_facts():
+    """Update the default set fact of hostname_facts setting with list of
+    facts like: bios_vendor,uuid
+
+    :id: 1042c5e2-ee4d-4eaf-a0b2-86c000a79dfb
+
+    :expectedresults: Default set fact should be updated with facts list.
+
+    :CaseAutomation: notautomated
+    """
+
 
-    @pytest.mark.stubbed
-    @tier2
-    def test_negative_update_hostname_with_empty_fact(self):
-        """Update the Hostname_facts settings without any string(empty values)
+@pytest.mark.stubbed
+@tier2
+def test_negative_discover_host_with_invalid_prefix():
+    """Update the hostname_prefix with invalid string like
+    -mac, 1mac or ^%$
+
+    :id: 73c5da42-28c8-4be7-8fb1-f505fefd6665
+
+    :expectedresults: Validation error should be raised on updating
+        hostname_prefix with invalid string, should start w/ letter
```

Comment:
w/ ?

Reply:
That's abbreviation for "with". "w/" = with, "w/o" = without.

But yeah, better to use full word, that's only 2 more characters.

Reply:
Fixed.